tui/src/driver.c

```diff
@@ -0,0 +1,95 @@
+#include "driver.h"
+
+#include <stdbool.h>
+#include <stdio.h>
+
+bool init_driver(driver_t *driver, const char *dev_path) {
+	bool init_success = false;
+
+	driver->serial_dev = NULL;
+	driver->serial_dev = fopen(dev_path, "wb");
+
+	if (driver->serial_dev == NULL) {
+		printf("Unable to open device at: %s\n", dev_path);
+	} else {
+		init_success = true;
+	}
+
+	return init_success;
+}
+
+void exit_driver(driver_t *driver) {
+	if (driver->serial_dev) {
+		fclose(driver->serial_dev);
+	} else {
+		printf("No device to close...\n");
+	}
+}
+
+/* This function arranges bytes into the Firmata/MIDI sequence needed. */
+static uint32_t format_firmata_bytes(int n_ubytes, uint8_t ubytes[n_ubytes]) {
```

Comment:
Wait did you do this because you didn’t realize you could send a char array instead of uint32_t or because you didn’t want to deal with char arrays????
Using uint32_t constrains this to only working on little endian systems but that probably doesn’t matter anyways and if this works then who gives a frick. It’s a demo anyways